airflow/models/__init__.py

```diff
@@ -2181,6 +2181,7 @@ def __init__(
         # Private attributes
         self._upstream_task_ids = set()
         self._downstream_task_ids = set()
+        self._rendered_template_object_ids = set()
```

Comment:
Ideally I'd like for this to not be a member. This is only used once, during template rendering, there's no need for having it here. The template rendering method could pass it along as parameter. This is not a big issue, it just seems cleaner to me to limit data to the scope where it's actually needed.

Reply:
I would rather keep it as a member to avoid changing `BaseOperator.render_template` and `BaseOperator.render_template_from_field` both signatures.

See `_render_template_object_recursively` implementation

Reply:
That's a fair point.